app/controllers/api/functions.php

```diff
@@ -1469,6 +1470,9 @@
             throw new Exception(Exception::DEPLOYMENT_NOT_FOUND);
         }
 
+        // buildId is added as param just for backward compatibility
+        // It is fetched from the deployments collection and not from the request as the user doesn't have to know the buildId
+        $buildId = $deployment->getAttribute('buildId', '');
         $build = Authorization::skip(fn () => $dbForProject->getDocument('builds', $buildId));
 
         if ($build->isEmpty()) {
```

Comment:
Why do we need this block of code? If you look at the code below, we don't use this build's ID for anything

Reply:
Removed the code as it wasn't being used anywhere